internal/blockchain/ethereum/ethereum.go

```diff
@@ -599,7 +665,22 @@ func (e *Ethereum) SubmitBatchPin(ctx context.Context, operationID *fftypes.UUID
 		batch.BatchPayloadRef,
 		ethHashes,
 	}
-	return e.invokeContractMethod(ctx, e.instancePath, signingKey, batchPinMethodABI, operationID.String(), input)
+	e.fireflyMux.Lock()
```

Comment:
I think you could drop the scope of this lock to be only:

```go
e.fireflyMux.Lock()
instancePath := e.instancePath
e.fireflyMux.Unlock()
```

Rather than holding it for the full invocation path. I understand that means the contract could change while the submission is in-flight. But I believe there is a much longer time window that exists for that anyway - the time (on all the nodes) to detect the blockchain event.